internal/vault/secrets/azure/azure_static_role_resource.go

```diff
@@ -0,0 +1,361 @@
+// refresh commands.
+//
+// https://developer.hashicorp.com/terraform/plugin/framework/resources/read
+func (r *AzureSecretsStaticRoleResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data AzureStaticRoleModel
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	cli, err := client.GetClient(ctx, r.Meta(), data.Namespace.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(errutil.ClientConfigureErr(err))
+		return
+	}
+
+	backend := data.Backend.ValueString()
+	role := data.Role.ValueString()
+	path := fmt.Sprintf("%s/%s/%s", backend, staticRolesAffix, role)
+
+	readResp, err := cli.Logical().ReadWithContext(ctx, path)
+	if err != nil {
+		resp.Diagnostics.AddError(errutil.VaultReadErr(err))
+		return
+	}
+	if readResp == nil {
+		resp.Diagnostics.AddError(errutil.VaultReadResponseNil())
+		return
+	}
+
+	var apiModel AzureStaticRoleAPIModel
+	err = model.ToAPIModel(readResp.Data, &apiModel)
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to translate Vault response data", err.Error())
+		return
+	}
+
+	// Map values back to Terraform model
+	data.ApplicationObjectID = types.StringValue(apiModel.ApplicationObjectID)
+	val, diags := types.MapValueFrom(ctx, types.StringType, apiModel.Metadata)
+
+	// Normalize TTL value since Vault can return either a string or a number
+	ttlSeconds, err := normalizeTTL(apiModel.TTL)
+	if err != nil {
+		resp.Diagnostics.AddError("Invalid TTL format from Vault", err.Error())
+		return
+	}
+	data.TTL = types.Int64Value(ttlSeconds)
+
+	resp.Diagnostics.Append(diags...)
+	data.Metadata = val
+	data.ID = types.StringValue(makeID(backend, role))
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *AzureSecretsStaticRoleResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data AzureStaticRoleModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	cli, err := client.GetClient(ctx, r.Meta(), data.Namespace.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(errutil.ClientConfigureErr(err))
+		return
+	}
+
+	backend := data.Backend.ValueString()
+	role := data.Role.ValueString()
+	path := fmt.Sprintf("%s/%s/%s", backend, staticRolesAffix, role)
+
+	vaultRequest, diags := buildVaultRequestFromModel(ctx, &data, false)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if _, err := cli.Logical().WriteWithContext(ctx, path, vaultRequest); err != nil {
+		resp.Diagnostics.AddError(errutil.VaultUpdateErr(err))
+		return
+	}
+
+	data.ID = types.StringValue(makeID(backend, role))
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func buildVaultRequestFromModel(ctx context.Context, data *AzureStaticRoleModel, includeSkipImport bool) (map[string]any, diag.Diagnostics) {
```

Comment:
Can we add a unit test for this function?

Reply:
I pulled this code out in a function, so I can reuse it for both Create and Update functions. I do have the test written for the workflow here: https://github.com/hashicorp/terraform-provider-vault/blob/8b2bb5d84a63503fe08f0170410389832e4849fa/internal/vault/secrets/azure/azure_static_role_resource_test.go#L26-L120

It tests both the initial config and then makes sure we can update it as well. Let me know if you had some other testing in mind here